core/merginapi.cpp

```diff
@@ -258,6 +260,43 @@ QString MerginApi::getApiKey( const QString &serverName )
   return "not-secret-key";
 }
 
+MerginConfig MerginApi::parseMerginConfig( const QString &projectDir )
+{
+  QString configFilePath( projectDir + sMerginConfigFile );
+
+  MerginConfig config;
+  QFile file( configFilePath );
+
+  if ( file.open( QIODevice::ReadOnly ) )
+  {
+    QByteArray data = file.readAll();
+    QJsonDocument doc = QJsonDocument::fromJson( data );
+    if ( doc.isObject() )
+    {
+      QJsonObject docObj = doc.object();
+      config.selectiveSyncEnabled = docObj.value( QStringLiteral( "input-selective-sync" ) ).toBool( false );
+      QString selectiveSyncDir = docObj.value( QStringLiteral( "input-selective-sync-dir" ) ).toString();
+
+      if ( !selectiveSyncDir.isEmpty() )
+      {
+        QDir rootDir( projectDir );
+        config.selectiveSyncDir = rootDir.canonicalPath() + "/" + selectiveSyncDir;
+      }
+    }
+    else
+    {
+      qDebug() << "MerginConfig: invalid content!";
+    }
+
+  }
+  else
+  {
+    qDebug() << "MerginConfig: cannot read a config file!";
```

Comment:
This will be printed for all projects that do not have config file, right? I was thinking about something less disturbing in that case like

```suggestion
    qDebug() << "MerginConfig: Project do not contain config file";
```
Or possibly no log at all